jms/1.8/modules/ROOT/pages/jms-troubleshooting.adoc

```diff
@@ -78,6 +78,130 @@ To resolve this scenario:
+
+JMS brokers implement prefetch policies for efficient usage of network resources using a push model to dispatch messages to consumers, ensuring that a consumer always has a local buffer of messages ready to process.
+When the JMS driver or broker has a prefetch size configuration, each consumer holds a buffer of messages ready to consume. The number of messages in the buffer depends on the broker implementation and custom parameterization.
+
+The JMS Connector *Consume* operation consumes one message at a time using a per operation consumer, which means that each operation holds a buffer of messages, but the operation consumes only the first message in the buffer by design. Which results in an unexpected behavior in the order in which messages are consumed. Consider the following scenarios:
+
+=== Concurrent Flows
+
+Conditions +
+
+* The same flow runs concurrently in two different threads of execution (thead A and thread B).
+* The JMS driver or broker prefetch size is configured to 10.
+* There are 100 messages in the queue.
+
+Flow iteration 1: +
+
+* Thread A: consumer A is created, and there are 10 messages in its local buffer that are ready to process and be pushed by the broker (messages 1 to 10).
+* Thread B: consumer B is created, and there are 10 messages in its local buffer that are ready to process that were pushed by the broker (messages 11 to 20 because the first 10 messages were pushed to consumer A).
+* Thread A: consumer A acknowledged message 1, and messages 2 to 10 were returned to the broker.
+* Thread B: consumer B acknowledged message 11, and messages 12 to 20 were returned to the broker.
+* Messages 1 and 11 were consumed.
+
+Flow iteration 2: +
+
+* Thread A: consumer A is created, and there are 10 messages in its local buffer that are ready to process and be pushed by the broker (messages 2,3,4,5,6,7,8,9,10,12).
+* Thread B: consumer B is created, and there are 10 messages in its local buffer that are ready to process that were pushed by the broker (messages 13 to 22).
+* Thread A: consumer A acknowledged message 2 and messages 3,4,5,6,7,8,9,10,12 were returned to the broker.
+* Thread B: consumer B acknowledged message 13 and messages 14 to 20 were returned to the broker.
+* Messages 2 and 13 were consumed.
+
+Flow iteration 3: +
+
+* Thread A: consumer A is created, and there are 10 messages in its local buffer that are ready to process and be pushed by the broker (messages 3,4,5,6,7,8,9,10,12,14).
+* Thread B: consumer B is created, and there are 10 messages in its local buffer that are ready to process that were pushed by the broker (messages 15 to 24).
+* Thread A: consumer A acknowledged message 3 and messages 4,5,6,7,8,9,10,12,14 were returned to the broker.
+* Thread B: consumer B acknowledged message 15 and messages 16 to 24 were returned to the broker.
+* Messages 3 and 15 were consumed.
+
+Workaround: +
```

Comment:
@konangc87: Is disabling prefetch a valid workaround for this scenario?

Reply:
Hi @nicomz , yes, disabling prefetch is another valid workaround.

Reply:
Updated @nicomz @konangc87 If all looks good, could you reapprove and I'll merge the PR. Thank you!